### DISCUSSION TITLE
SEB 3.1.1 for Mac does not prevent app switching or copy paste?

### DISCUSSION THREAD

**Discussion by ben-teaches-english** [SafeExamBrowser/seb-mac#252] [General]
I used manual configuration within a Moodle quiz and a student on their Macbook was able to switch to his password manager and copy paste his password to log into Moodle! Is this a setting within the manual configuration or is this an issue with SEB?

**Comment by danschlet** [comment]
This should definitely not be possible with an unmanipulated SEB using unmanipulated Moodle manual configuration. 

Can you check if that student used the original SEB version? Use [SEB Verificator](https://github.com/SafeExamBrowser/seb-mac-verificator/releases) and make sure the student doesn't have other SEB or SEB-alike versions installed (which the Verificator would show). Then watch the student starting the quiz from another browser (so they really start it with the original SEB).

Otherwise please find out what password manager they used exactly and ideally film this switch to the password manager and send us the video.

**Reply by ben-teaches-english** [reply]
Thank you for the quick reply! I will see this student next week and try to replicate it. 

I don't suspect malicious intent (famous last words, I know), I believe the student got their version via the "Download SEB" button in the Moodle quiz which goes to this page: https://safeexambrowser.org/download_en.html

**Reply by ben-teaches-english** [reply]
'Progress' report: I tested positive for COVID at the weekend so was unable to see my students in person this week. I asked them to record the software verification and test for me but they couldn't share the video and said they were unable to repeat the window switching. They also said the Verificator returned an invalid signature. I'll try talking to them next week to get some more useful data. Thanks for your patience.

**Reply by danschlet** [reply]
I realized that the current version of the SEB Verificator doesn't recognize SEB 3.1.1 (and 3.2.1 which I'm in the process of releasing) correctly because of a changed code signing certificate, I'll release an update in the next days.

Get well soon!

**Comment by ben-teaches-english** [comment]
Hi again, 

New year, same problem but with new students and the updated version of SEB.

Here is the behaviour we found when testing SEB on an iPad. She is a new student and downloaded from the "Download Safe Exam Browser" button in Moodle. I am using a default config that works as desired on Windows and Mac laptops.

[https://photos.app.goo.gl/zCHv6H8ANSaGa2dt8](https://photos.app.goo.gl/zCHv6H8ANSaGa2dt8) 

Is there a setting I need to change somewhere? Thank you for your help again with this.

**Reply by ben-teaches-english** [reply]
Here's another video of a student using version 3.3 and able to open DeepL while in SEB and minimize the SEB window. Copy/paste is still disabled, but he was able to use the other apps just fine otherwise. 

Download link for the video (it's about 138MB) https://www.swisstransfer.com/d/3f528d25-20b5-4fb6-a6d8-626aae25082d

**Reply by danschlet** [reply]
You need to set a quit password, only then SEB for iOS locks the device. It also doesn't make sense at all to use SEB for Windows or macOS without quit password, because then students can quit SEB any time, use the internet and then re-enter the exam (Moodle for example doesn't prevent that).

Of course you need to remember that quit password, otherwise you cannot unlock the device in case something goes wrong with the quiz/Moode [server](https://moodle.org/mod/glossary/showentry.php?eid=30&displayformat=dictionary)/internet connection.

See [SEB manual](https://safeexambrowser.org/ios/ios_usermanual_en.html) for more information.

**Reply by ben-teaches-english** [reply]
Thank you so much for the help. I just hadn't made the connection between the behaviour I wanted and the quit password/quit link terminology. It also behaved like I wanted on my test device but I wasn't devious enough to think about quitting and reloading so I'm glad you anticipated that! I've read the manual more carefully now and I've updated the settings for another practice task tomorrow. Thanks again for your patience and assistance!

**Comment by aurelienb1** [comment]
Hi, I'm not sure if it's the same behavior, but take a look at this moodle tracker: https://tracker.moodle.org/browse/MDL-79968. 

The "Launch SEB" button is different from "Donwload configuration".

Regards

**Reply by danschlet** [reply]
I don't seem to have permission to view that issue in Moodle tracker (even if I'm logged in).

**Comment by aurelienb1** [comment]
Hi here the summary

Currently, there are two ways to start an exam:

Launch safe exam browser
Download the configuration

If the student uses "Launch SEB", then he won't be using the template, but SEB in its simplest form. The point here is to be able to bypass the template's highest security level. Launching the secure exam browser with this buton enables more things to be done in SEB, such as opening additional windows in SEB to browse the course (this is a real-life situation).
It is therefore important to be able to control the display of the "Launch" button, just like the download button. So when a template is used, only this option needs to be displayed.

Regards

**Reply by danschlet** [reply]
Maybe someone can add me (username danschlet) to the Moodle tracker issue. I don't really understand the issue, but maybe our Moodle developer will know what's going on, as he knows more details about the SEB Moodle integration.

**Reply by aurelienb1** [reply]
In fact, in moodle, the "Launch SEB" button starts SEB with the bare minimum (as long as you haven't distributed any files to configure the clients). Whereas the download configuration button uses a SEB template that you can provide (which is much more complete, such as domain restriction or other security or configuration elements).

I don't know if this helps as an explanation.

**Reply by danschlet** [reply]
Not really. Do you mean "bare minimum" settings? That would be irrelevant, because SEB always uses (secure) default values for configuration keys which are not contained in a configuration file.

**Reply by aurelienb1** [reply]
I aded you like "peer revievew" in this moodle task

**Reply by aurelienb1** [reply]
Over time I realized that some students were able to exit seb via links in the WEB interface (open other windows in SEB or consult their course) when they just used "Launch SEB".

**Reply by danschlet** [reply]
That's another issue, which will be fixed in the 3.3.1 final release.

**Reply by danschlet** [reply]
Unfortunately I still can't view the Moodle Tracker issue.
